pkg/vpa/test_constants.go

```diff
@@ -74,3 +84,12 @@ var testDeployment = &appsv1.Deployment{
 		Name: "test-deploy",
 	},
 }
+
+var testDeploymentExcluded = &appsv1.Deployment{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "test-deploy",
+		Annotations: map[string]string{
```

Comment:
Curious why you chose to do annotations on the deployments when we use labels on the namespace? This seems like it might be confusing for the end-users.

Reply:
The whole labels vs. annotations thing is something I'm not quite sure of in terms of best practices from a kubernetes stand point too. From my understanding labels should be used sparingly to be used in a selector manner, and annotations are more free form in their purpose. I form this opinion mostly from here: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/

> You can use either labels or annotations to attach metadata to Kubernetes objects. Labels can be used to select objects and to find collections of objects that satisfy certain conditions. In contrast, annotations are not used to identify and select objects. The metadata in an annotation can be small or large, structured or unstructured, and can include characters not permitted by labels.

So in all honesty I'm not even sure we should have vpa mode or other things as labels. Overall it's your product and decision though :)

Reply:
I think you're probably right about annotations vs labels. However at this point we have gone all in on labels, and now is probably not the time to change this.  

there's probably a world where we switch to using annotations, but it will require supporting both for a time I think

Reply:
As a team we have decided that long-term annotations are the way to go. 

With that in mind, I opened #148 to track that. We will implement a graduated deprecation to do that.

For now, the deployments using annotations is fine.

Reply:
Great! I didn't mean to move y'all's cheese, sorry about that :)

Reply:
Not a problem. I think it actually came up before, but we punted.